File: 1. Array & Hashing/238. 除自身以外数组的乘积/238. 除自身以外数组的乘积.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1]*len(nums)
        temp1 = 1
        for i in range(len(nums)):
            res[i] = res[i]*temp1
            temp1 = temp1*nums[i]
        # print(res)
        temp2 = 1
        for j in range(len(nums)-1,-1,-1):  
            res[j] = res[j]*temp2
            temp2 = temp2*nums[j]
        return res

File: 1. Array & Hashing/238. 除自身以外数组的乘积/238. 除自身以外数组的乘积.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n,1);
        int temp1 = 1;
        int temp2 = 1;
        for(int i = 0; i < n; i++){
            cout<< i <<endl;
            res[i] = res[i]*temp1;
            temp1 = temp1*nums[i];
        }
        for(int j = n-1; j >= 0; j--){
            res[j] = res[j]*temp2;
            temp2 = temp2*nums[j];
        }
        return res;
    }
};